File: backend/src/main/java/com/subman/submanapi/model/Subscription.java
package com.subman.submanapi.model;

import com.subman.submanapi.enums.BillingCycle;
import com.subman.submanapi.enums.Category;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.LinkedList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Subscription {
    private Long id;
    private Long userId;
    private String nome;
    private BigDecimal valor;
    private Category categoria;
    private BillingCycle cicloCobranca;
    private LocalDate vencimento;
    private List<String> historicoPagamentos = new LinkedList<>();
}

File: frontend/src/types/subscription.ts
// Enum para Categoria
export enum Category {
  STREAMING = "STREAMING",
  GAMES = "GAMES",
  MOBILE = "MOBILE",
  SERVICE = "SERVICE",
  SAAS = "SAAS",
  OTHER = "OTHER",
}

// Enum para Ciclo
export enum BillingCycle {
  WEEKLY = "WEEKLY",
  MONTHLY = "MONTHLY",
  QUARTERLY = "QUARTERLY",
  SEMI_ANNUAL = "SEMI_ANNUAL",
  ANNUAL = "ANNUAL",
}

// O DTO da Assinatura
export interface Subscription {
  id: number;
  userId: number;
  nome: string;
  valor: number;
  categoria: Category;
  cicloCobranca: BillingCycle;
  vencimento: string; // string (ISO date)
  historicoPagamentos: string[];
}